src/transformers/models/oneformer/modeling_oneformer.py

```diff
@@ -2801,6 +2801,7 @@ def _init_weights(self, module: nn.Module):
         elif isinstance(module, OneFormerTransformerDecoder):
             nn.init.xavier_uniform_(module.query_input_projection.weight, gain=xavier_std)
             nn.init.constant_(module.query_input_projection.bias, 0)
+            module.query_input_projection._is_hf_initialized = True
```

Comment:
once all instances of `OneFormerTransformerDecoder` submodule will have `_is_hf_initialized = True` this code will already never run, no? as `_init_weights` won't get called on this sub-module anymore.

Reply:
The goal with this is to avoid the `module.query_input_projections` be initialized another time since it is a Conv2d and there is a path for Conv2d in the succession of tests here. This is an example of fix for problem 2 in the PR description.

Reply:
Got it. I can see now why:
```
elif isinstance(module, (nn.Linear, nn.Conv2d, nn.BatchNorm2d)):
```

Reply:
Here is an idea - pass param key name in addition to module name where possible? that way one could also `if` on the key name and shortcut the "switch case"

Reply:
so instead of - init Conv2d this way unless the param belongs to this parent module - for this param key only use this init.

Reply:
but I guess it'd be difficult if the key isn't always fully qualified - perhaps just checking the last segment?

Reply:
That would require changing the signature of all `_init_weights`, so we're back to changing all models ;-) I think it's probably easier this way even if it looks more convoluted at first glance.

Reply:
ah, didn't think of that one! you're correct, Sylvain.

It'd have been useful if the init function returns a list of params it touched. the the outside care taker could do the accounting automatically. But again this adds more complexity.

Reply:
This reminds me of the Deepspeed external parameter special case that was originally an issue for the same reason.

So let's please document this special case using the full example with `2 isinstance` branches to show what to do with a sub-module inits weights that are outside of its immediate descendants.

Reply:
Will add it to the add model doc.